internal/security/credentials.go

```diff
@@ -47,30 +53,81 @@ func (s *SecretProvider) GetDatabaseCredentials(database db.DatabaseInfo) (commo
 // specified path will be returned.
 func (s *SecretProvider) GetSecrets(path string, keys ...string) (map[string]string, error) {
 
+	if cachedSecrets := s.getCachedSecrets(path, keys...); cachedSecrets != nil {
+		return cachedSecrets, nil
+	}
+
+	if !s.isSecurityEnabled() {
+		s.loggingClient.Debug(fmt.Sprintf("Writable.InsecureSecrets: %v", s.insecureSecrets))
+		return s.getInsecureSecrets(path, keys...)
+	}
+
+	newSecrets, err := s.secretClient.GetSecrets(path, keys...)
+	if err != nil {
+		return nil, err
+	}
+
+	s.secretsCache[path] = newSecrets
+	return s.secretsCache[path], nil
```

Comment:
Should return `newSecrets` once `newSecrets` is add to the cache rather then overriding the cache.